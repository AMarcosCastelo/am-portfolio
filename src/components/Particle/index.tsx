import Particles from '../../../node_modules/react-particles-js'

import particleConfigDark from '../../config/particlesjs-config.json'
import particleConfigLight from '../../config/particlesjs-config-light.json'

import * as S from './styles'

type Props = {
  theme: 'light' | 'dark'
}

const Particle: React.FC<Props> = ({ theme }: Props) => (
  <S.Wrapper>
    <Particles
      // @ts-ignore: params type
      params={theme === 'light' ? particleConfigLight : particleConfigDark}
    />
  </S.Wrapper>
)

export default Particle
