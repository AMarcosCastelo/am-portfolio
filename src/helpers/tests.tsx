import { ThemeProvider } from 'styled-components'
import { render, RenderResult } from '@testing-library/react'

import { LIGHT_THEME, DARK_THEME } from 'styles'

export const renderWithTheme = (
  children: React.ReactNode,
  theme?: 'light' | 'dark'
): RenderResult =>
  render(
    <ThemeProvider theme={theme === 'dark' ? DARK_THEME : LIGHT_THEME}>
      {children}
    </ThemeProvider>
  )
