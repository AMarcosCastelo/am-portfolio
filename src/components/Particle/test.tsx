import { renderWithTheme } from 'helpers'

import Particle from '.'

describe('<Particle />', () => {
  it('should render the Particle', () => {
    const { container } = renderWithTheme(<Particle theme="light" />)

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render the Particle with dark theme', () => {
    const { container } = renderWithTheme(<Particle theme="dark" />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
