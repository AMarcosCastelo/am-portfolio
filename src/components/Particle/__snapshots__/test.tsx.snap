// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Particle /> should render the Particle 1`] = `
.c0 > #tsparticles {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 0;
}

<div
  class="c0"
>
  <div
    id="tsparticles"
  >
    <canvas
      class="tsparticles-canvas-el"
      height="0"
      style="width: 100%; height: 100%;"
      width="0"
    />
  </div>
</div>
`;

exports[`<Particle /> should render the Particle with dark theme 1`] = `
.c0 > #tsparticles {
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  z-index: 0;
}

<div
  class="c0"
>
  <div
    id="tsparticles"
  >
    <canvas
      class="tsparticles-canvas-el"
      height="0"
      style="width: 100%; height: 100%;"
      width="0"
    />
  </div>
</div>
`;
